src/lib/menu/menu-trigger.ts

```diff
@@ -181,14 +183,33 @@ export class MdMenuTrigger implements AfterViewInit, OnDestroy {
    * @returns ConnectedPositionStrategy
    */
   private _getPosition(): ConnectedPositionStrategy  {
-    const positionX: HorizontalConnectionPos = this.menu.positionX === 'before' ? 'end' : 'start';
-    const positionY: VerticalConnectionPos = this.menu.positionY === 'above' ? 'bottom' : 'top';
-
-    return this._overlay.position().connectedTo(
-      this._element,
-      {originX: positionX, originY: positionY},
-      {overlayX: positionX, overlayY: positionY}
-    );
+    const [posX, fallbackX]: HorizontalConnectionPos[] =
+      this.menu.positionX === 'before' ? ['end', 'start'] : ['start', 'end'];
+
+    const [posY, fallbackY]: VerticalConnectionPos[] =
+      this.menu.positionY === 'above' ? ['bottom', 'top'] : ['top', 'bottom'];
+
+    return this._overlay.position()
+      .connectedTo(this._element, this._originPos(posX, posY), this._overlayPos(posX, posY))
+      .withFallbackPosition(
+        this._originPos(fallbackX, posY), this._overlayPos(fallbackX, posY))
+      .withFallbackPosition(
+        this._originPos(posX, fallbackY), this._overlayPos(posX, fallbackY))
+      .withFallbackPosition(
+        this._originPos(fallbackX, fallbackY), this._overlayPos(fallbackX, fallbackY));
+  }
+
+
+  /**  Converts the designated point into an OriginConnectionPosition. */
+  private _originPos(x: HorizontalConnectionPos,
+                     y: VerticalConnectionPos): OriginConnectionPosition {
+    return {originX: x, originY: y} as OriginConnectionPosition;
```

Comment:
Is the `as OriginConnectionPosition` necessary? The types should be structurally identical.